Add more context to the Traccar Server documentation (#31207)
home-assistant/home-assistant.github.io: source/_integrations/traccar_server.markdown

</div>

{% include integrations/config_flow.md %}

## Events

> The Traccar Server integration options allows you to define a list of events that you want to listen for. These events will be sent to Home Assistant as [events](/docs/configuration/events/).

Reviewer: ```suggestion
The Traccar Server integration options allow you to define a list of events you want to listen for. These events will be sent to Home Assistant as [events](/docs/configuration/events/).
```

All events will be prefixed with `traccar_` followed by a snake_cased version of the event name.
Examples:

- The `deviceMoving` will be fired in Home Assistant as `traccar_device_moving`.
- The `geofenceExit` will be fired in Home Assistant as `traccar_geofence_exit`.
- The `alarm` will be fired in Home Assistant as `traccar_alarm`.

All events will also have the following data:

{% configuration_basic %}

Device ID:
  description: (`device_traccar_id`) This will be the device ID that the event is related to.
Device name:
  description: (`device_name`) This will be the name of the device that the event is related to.
Type:
  description: (`type`) This will be the type of the event.
Server time:
  description: (`serverTime`) This will be the time the event was received by the Traccar Server.
Attributes:
  description: (`attributes`) This will be a dictionary of attributes related to the event.

{% endconfiguration_basic %}

{% details "Example" %}

```json
{
    "device_traccar_id": 24,
    "device_name": "My Device",
    "type": "alarm",
    "serverTime": "1970-01-01T00:00:00.000Z",
    "attributes": {
      "custom_attribute": "value"
    },
}
```

{% enddetails %}

## Devices

> The Traccar Server integration will create devices for each device registered in Traccar Server that has at least one position update.

Reviewer: ```suggestion
The Traccar Server integration will create devices for each device registered in the Traccar Server with at least one position update.
```

> These device representations in Home Assistant will have [entities](#entities) associated with them, which you can use in [automations](/docs/automation), [scripts](/docs/script) and display on your [dashboard](/docs/dashboard).

Reviewer: ```suggestion
These device representations in Home Assistant will have [entities](#entities) associated with them, which you can use in [automations](/docs/automation), [scripts](/docs/script), and display on your [dashboard](/docs/dashboard).
```

### Device Tracker

The Traccar Server integration will create a [device tracker](/integrations/device_tracker) entity for each device registered in Traccar Server.

{% configuration_basic %}
Name:
  description: The name of the device tracker will be set to what you have named it in Traccar Server.
Entity ID:
  description: This will be a slugified version of the name of the device tracker.
Unique ID:
  description: This will be the unique ID of the device tracker in Traccar Server.
State:
  description: As a device tracker, the state will be `home` or `not_home`.
{% endconfiguration_basic %}

{% details "Attributes" %}

In addition to the custom attributes you can define in the Traccar Server integration options, the device tracker entity will have the following attributes:

{% configuration_basic %}
Address:
  description: If a position update has an address associated with it, this will be the address.
Altitude:
  description: The altitude of the position update.
Battery Level:
  description: The battery level of the device if defined.
Category:
  description: The category of the device in Traccar if defined.
Geofence:

>   description: The name of the geofence the device located in.

Reviewer: ```suggestion
  description: The name of the geofence the device is located in.
```
